salt/modules/boto_iam.py

```diff
@@ -153,6 +153,297 @@ def role_exists(name, region=None, key=None, keyid=None, profile=None):
+    '''
+    conn = _get_conn(region, key, keyid, profile)
+    try:
+        info = conn.get_role(name)
+        if not info:
+            return False
+        return info
+    except boto.exception.BotoServerError as e:
+        log.debug(e)
+        msg = 'Failed to get {0} information.'
+        log.error(msg.format(name))
+        return False
+
+
+def create_user(user_name, path=None, region=None, key=None, keyid=None,
+                profile=None):
+    '''
+    Create a user
+
+    CLI example::
+
+        salt myminion boto_iam.create_user myuser
+    '''
+    if not path:
+        path = '/'
+    if __salt__['boto_iam.get_user'](user_name, region, key, keyid, profile):
+        return True
+    conn = _get_conn(region, key, keyid, profile)
+    try:
+        conn.create_user(user_name, path)
+        log.info('Created user : {0} .'.format(user_name))
+        return True
+    except boto.exception.BotoServerError as e:
+        log.debug(e)
+        msg = 'Failed to create user {0} .'
+        log.error(msg.format(user_name))
+        return False
+
+
+def get_user(user_name=None, region=None, key=None, keyid=None, profile=None):
+    '''
+    Get user information
+
+    CLI example::
+
+        salt myminion boto_iam.get_user myuser
+    '''
+    conn = _get_conn(region, key, keyid, profile)
+    try:
+        info = conn.get_user(user_name)
+        if not info:
+            return False
+        return info
+    except boto.exception.BotoServerError as e:
+        log.debug(e)
+        msg = 'Failed to get user {0} info .'
+        log.error(msg.format(user_name))
+        return False
+
+
+def create_group(group_name, path=None, region=None, key=None, keyid=None,
+                 profile=None):
+    '''
+    Create a group
+
+    CLI example::
+
+        salt myminion boto_iam.create_group group
+    '''
+    if not path:
+        path = '/'
+    if __salt__['boto_iam.get_group'](group_name=group_name, region=region, key=key, keyid=keyid, profile=profile):
+        return True
+    conn = _get_conn(region, key, keyid, profile)
+    try:
+        conn.create_group(group_name, path)
+        log.info('Created group : {0} .'.format(group_name))
+        return True
+    except boto.exception.BotoServerError as e:
+        log.debug(e)
+        msg = 'Failed to create group {0} .'
+        log.error(msg.format(group_name))
+        return False
+
+
+def get_group(group_name, marker=None, max_items=None, region=None, key=None,
+              keyid=None, profile=None):
+    '''
+    Get group information
+
+    CLI example::
+
+        salt myminion boto_iam.get_group mygroup
+    '''
+    conn = _get_conn(region, key, keyid, profile)
+    try:
+        info = conn.get_group(group_name)
+        if not info:
+            return False
+        return info
+    except boto.exception.BotoServerError as e:
+        log.debug(e)
+        msg = 'Failed to get group {0} info .'
+        log.error(msg.format(group_name))
+        return False
```

Comment:
You could throw away a lot of code lines, if this and similar places all around everywhere could be written as following (or even wrapped into a decorator):
```python
try:
    return  _get_conn(region, key, keyid, profile).get_group(group_name) or False
except: boto.exception.BotoServerError as e:
    log.debug(e)
    log.error('Failed to get group {0} info .'.format(group_name))
    return False
```